lib/ts/types.ts

```diff
@@ -30,14 +30,15 @@ export type SuperTokensConfig = {
     /*
      * List of recipes for authentication and session management.
      */
-    recipeList: CreateRecipeFunction<any, any, any>[];
+    recipeList: CreateRecipeFunction<any, any, any, any>[];
 };
 
 export type CreateRecipeFunction<
     Action,
     PreAPIHookContext extends RecipePreAPIHookContext<Action>,
-    Config extends RecipeConfig<Action, PreAPIHookContext>
-> = (appInfo: NormalisedAppInfo) => RecipeModule<Action, PreAPIHookContext, Config>;
+    PostAPIHookContext extends RecipePostAPIHookContext<Action>,
+    Config extends RecipeConfig<Action, PreAPIHookContext, PostAPIHookContext>
+> = (appInfo: NormalisedAppInfo) => RecipeModule<Action, PreAPIHookContext, PostAPIHookContext, Config>;
```

Comment:
The only variable in `PreAPIHookContext` and `PostAPIHookContext` from a types point of view is `Action`. So you can make this `RecipeModule<Action, PreAndPostAPIHookContextAction, Config>`

Reply:
Done, changed to `RecipeModule<Action, RecipeConfig<Action>>`

Reply:
You can make `RecipeModule<Action, RecipeConfig<Action>>` into `RecipeModule<Action>`

Reply:
Done